backend/endpoints/connections.js

```diff
@@ -103,103 +103,95 @@ module.exports = async function (app, createOctokit, connectionsGraph) {
   }
 
   /***
-   * Compares the similarities of the GitHubs of two users.
-   * Considers and weighs factors such as code breakdown of repositories, tags of repositories, etc.
-   * Uses the octokit of the user who the recommendation is being used for.
+   * Gets GitHub information, including repository information.
    */
-  async function compareGitHubs(
-    userOneGitHubHandle,
-    userTwoGitHubHandle,
-    octokit
-  ) {
-    /***
-     * Gets GitHub information, including repository information.
-     */
-    async function getGitHubInfo(githubHandle) {
-      const repositoryResponse = await octokit.request(
-        "GET /users/{owner}/repos",
+  async function getGitHubInfo(githubHandle, octokit) {
+    const repositoryResponse = await octokit.request(
+      "GET /users/{owner}/repos",
+      {
+        sort: "pushed",
+        owner: githubHandle,
+      }
+    );
+    const repositories = repositoryResponse.data;
+
+    let languages = {}; // Cumulative language makeup of recent repositories.
+    const topics = [];
+    let repositoriesIndex = 0;
+
+    // Looks at the 5 most recent repositories for analysis of languages and topics.
+    while (
+      repositoriesIndex < repositories.length &&
+      repositoriesIndex < ANALYZE_REPOS
```

Comment:
tip: instead of having two different conditions, you can do `while (repositoriesIndex < min(ANALYZE_repos, repositories.length))`

Reply:
That's smart! Thank you!!